pkg/cmd/pulumi/package.go

```diff
@@ -195,17 +229,23 @@ func providerFromSource(packageSource string) (plugin.Provider, error) {
 		return provider, nil
 	}
 
-	// We were given a path to a binary, so invoke that.
+	// We were given a path to a binary or folder, so invoke that.
 	info, err := os.Stat(pkg)
 	if os.IsNotExist(err) {
 		return nil, fmt.Errorf("could not find file %s", pkg)
 	} else if err != nil {
 		return nil, err
-	} else if !isExecutable(info) {
-		if p, err := filepath.Abs(pkg); err == nil {
-			pkg = p
+	} else if info.IsDir() {
+		// If it's a directory we need to add a fake provider binary to the path because that's what NewProviderFromPath
+		// expects.
+		pkg = filepath.Join(pkg, "pulumi-resource-provider")
+	} else {
```

Comment:
I don't understand this change. It seems
a. unrelated to supporting parametrize
b. introduces ambiguity when a directory is selected: how is the actual binary selected
c. necessary to document this in the public `--help` section for `get-schema` if we take the change.

Reply:
> unrelated to supporting parametrize

Aye it is unrelated, it's just testing this with shimless is the easiest way to test this.

>  introduces ambiguity when a directory is selected: how is the actual binary selected

Slightly... as long as the filename we pick doesn't exist then ProviderFromPath will run the folder (it does an Exists check)

> necessary to document this in the public

Yes, I'll add that.